kafka-client/src/main/java/dev/responsive/internal/stores/CommitBuffer.java

```diff
@@ -274,44 +273,50 @@ private boolean shouldFlush() {
     boolean bytesTrigger = false;
     boolean timeTrigger = false;
     if (buffer.getReader().size() >= flushTriggers.getRecords()) {
-      log.info("will flush due to records buffered {} over trigger {}",
+      log.debug("Will flush due to records buffered {} over trigger {}",
           buffer.getReader().size(),
           flushTriggers.getRecords()
       );
       recordsTrigger = true;
     } else {
-      log.debug("records buffered {} not over trigger {}",
+      log.debug("Records buffered since last flush {} not over trigger {}",
           buffer.getReader().size(),
           flushTriggers.getRecords()
       );
     }
     final long totalBytesBuffered = totalBytesBuffered();
     if (totalBytesBuffered >= flushTriggers.getBytes()) {
-      log.info("will flush due to bytes buffered {} over bytes trigger {}",
+      log.debug("Will flush due to bytes buffered {} over bytes trigger {}",
           totalBytesBuffered,
           flushTriggers.getBytes()
       );
       bytesTrigger = true;
     } else {
-      log.debug("bytes buffered {} not over trigger {}",
+      log.debug("Bytes buffered since last flush {} not over trigger {}",
           totalBytesBuffered,
           flushTriggers.getBytes()
       );
     }
     final var now = clock.get();
     if (lastFlush.plus(flushTriggers.getInterval()).isBefore(now)) {
-      log.info("will flush due to time since last flush {} over interval trigger {}",
+      log.debug("Will flush as time since last flush {} over interval trigger {}",
           Duration.between(lastFlush, now),
           now
       );
       timeTrigger = true;
     } else {
-      log.debug("time since last flush {} not over trigger {}",
+      log.debug("Time since last flush {} not over trigger {}",
           Duration.between(lastFlush, now),
           now
       );
     }
-    return recordsTrigger || bytesTrigger || timeTrigger;
+    final boolean shouldFlush = recordsTrigger || bytesTrigger || timeTrigger;
+
+    if (shouldFlush) {
+      log.info("Triggering flush due to recordsTrigger={}, bytesTrigger={}, timeTrigger={}",
```

Comment:
we might as well add the number of records we're about to flush here (and the current topic-partition/virtual-partition/offset) - I think all of those are helpful when debugging.

Reply:
Fair enough -- was on the fence about whether or not to leave even one INFO level log per flush and added this back at the last minute, but if there's anything logged per flush at INFO it should at least have all the useful information.

I'll remove this one entirely and then log the summary without the offset interval logic. Also just to clarify up front in case you were going to ask, I'm intentionally leaving out the kafka topic/partition because that's static and is included automatically in the logger's prefix.